.github/workflows/pull-request.yaml

```diff
@@ -117,10 +117,12 @@ jobs:
         include:
           - os: ubuntu-latest
             target: x86_64-unknown-linux-musl
-          - os: ubuntu-latest
-            target: aarch64-unknown-linux-musl
-          - os: macos-latest
-            target: x86_64-apple-darwin
+          # TODO: Until we have tests for these, we don't have a cache for them.
+          # If we can add tests, then re-enable them. They run on `release.yaml`
+          # regardless.
+          #
+          # - os: ubuntu-latest target: aarch64-unknown-linux-musl
+          # - os: macos-latest target: x86_64-apple-darwin
```

Comment:
My understanding is that `x86_64-apple-darwin` is tested on GitHub Actions and `aarch64-apple-darwin` is not.
Is it really necessary to disable these?

Reply:
Do you think we could add these to the `test-all` or `nightly` tests? Then they'll share the cache and be v fast.

We can only use this new approach of cache-sharing if they're also run in tests, because the caches are only written in tests. Or we could a) revert — though this way will lead to far fewer cache misses, or b) do something like a label, where it's longer, but not required on every PR, or c) have them here but they'll take a long time.

Reply:
I hadn't realized that Mac was on x86_64. I'll switch it (doesn't affect the discussion above)

Reply:
> Do you think we could add these to the `test-all` or `nightly` tests? Then they'll share the cache and be v fast.

I think it makes sense to add it to nightly. (In short, one of the purposes of this is to prevent people from realizing that their builds are broken only at release time, so I think we should reduce the number of targets that are only built at release time.)

I am not sure I fully understand your explanation, but are you assuming that if we add this to nightly we can re-enable the build in PR?

Reply:
> I am not sure I fully understand your explanation, but are you assuming that if we add this to nightly we can re-enable the build in PR?

Yes!

Reply:
Ok, thanks. I'll take a look at later!

Reply:
The only thing to watch for is that they should use a different key, and I'm not sure how the toolchain gets into the key; maybe setting it with `rustup` before running the cache action will cause it to be different....